lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -199,12 +211,17 @@ def event_action_params(event)
     end
 
     # Determine the correct value for the 'type' field for the given event
+    DEFAULT_EVENT_TYPE="doc".freeze
     def get_event_type(event)
       # Set the 'type' value for the index.
       type = if @document_type
                event.sprintf(@document_type)
              else
-               event.get("type") || "doc"
+               if maximum_seen_major_version > 6
+                 event.get("type") || DEFAULT_EVENT_TYPE
```

Comment:
I'm trying to understand this code, and for ES 6.0 it seems like it to still try to set the type based on the event type? Can't each event still have multiple different types?

Reply:
Ah, good call, the logic is wrong, that should be `<= 6`

Reply:
Errr, `< 6` I mean